scrapers/nd/bills.py

```diff
@@ -1,229 +1,161 @@
-                'or contains(@summary, "Resolutions")'
-                "]//tr/th/a/@href"
+
+class BillDetail(HtmlPage):
+    input_type = Bill
+    example_input = Bill(
+        "HB 1001",
+        "2021",
+        "[title]",
+        chamber="lower",
+        classification="bill",
+    )
+
+    def get_source_from_input(self):
+        return self.input.sources[0]["url"]
+
+    def process_page(self):
+        self.process_versions()
+        self.process_actions()
+        yield self.input
+
+    def process_versions(self):
+        source = re.sub("overview/bo", "index/bi", self.source.url)
+        response = requests.get(source)
+        content = lxml.html.fromstring(response.content)
+
+        version_rows = content.xpath(".//table[@id='version-table']/tbody//tr")
+        for row in version_rows:
+            vers_links = row.xpath("td[1]//a")
+
+            for link in vers_links:
+                name = row.xpath("td[2]/text()")
+                if name:
+                    (name,) = name
+                else:
+                    (name,) = link.xpath("text()")
+
+                type_badge = link.xpath("span[@data-toggle='tooltip']")
+                if type_badge:
+                    (vers_type,) = type_badge[0].xpath("@title")
+                    if vers_type.strip() == "Marked up":
+                        name += "(Marked up)"
+                else:
+                    vers_type = None
+
+                (href,) = link.xpath("@href")
+                url = re.sub("/bill-index.+", href[2:], source)
+
+                if not vers_type or vers_type.strip() == "Engrossment":
+                    self.input.add_version_link(note=name, url=url, media_type="pdf")
+                else:
+                    self.input.add_document_link(note=name, url=url, media_type="pdf")
+
+    def process_actions(self):
+        source = re.sub("overview/bo", "actions/ba", self.source.url)
+        response = requests.get(source)
+        content = lxml.html.fromstring(response.content)
+
+        action_rows = content.xpath(".//table[@id='action-table']/tbody//tr")
+        for row in action_rows:
+            (action,) = row.xpath("td[3]/text()")
+
+            actor = "legislature"
+            chambers_dict = {"House": "lower", "Senate": "upper"}
+            chamber = row.xpath("td[2]/text()")
+            if chamber:
+                chamber = chamber[0].strip()
+                if chamber in chambers_dict.keys():
+                    actor = chambers_dict[chamber]
+            if "governor" in action.lower():
+                actor = "executive"
+
+            (date,) = row.xpath("td[1]/b/text()")
+            date += f"/{self.input.legislative_session}"
+            date = dt.datetime.strptime(date, "%m/%d/%Y")
+
+            self.input.add_action(
+                action,
+                date.strftime("%Y-%m-%d"),
+                chamber=actor,
+                classification="introduction",
```

Comment:
Make sure to use the action classifier here!

Reply:
Of course, I forgot to take out that placeholder. [Here it is added in](https://github.com/openstates/openstates-scrapers/pull/4190/commits/765499233afd0d3168d57142cb4b5a039c6f6d67#diff-1ada3395bc035a7192d515167751ad1b19db1e3ef3b2b7d4ae9052d41a75b15cR156).